server/schedule/filter/filters.go

```diff
@@ -492,3 +493,78 @@ func (f *BlacklistStoreFilter) filter(store *core.StoreInfo) bool {
 	_, ok := f.blacklist[store.GetID()]
 	return ok
 }
+
+// labelConstraintFilter is a filter that selects stores satisfy the constraints.
+type labelConstraintFilter struct {
+	scope       string
+	constraints []placement.LabelConstraint
+}
+
+// NewLabelConstaintFilter creates a filter that selects stores satisfy the constraints.
+func NewLabelConstaintFilter(scope string, constraints []placement.LabelConstraint) Filter {
+	return labelConstraintFilter{scope: scope, constraints: constraints}
+}
+
+// Scope returns the scheduler or the checker which the filter acts on.
+func (f labelConstraintFilter) Scope() string {
+	return f.scope
+}
+
+// Type returns the name of the filter.
+func (f labelConstraintFilter) Type() string {
+	return "label-constraint-filter"
+}
+
+// Source filters stores when select them as schedule source.
+func (f labelConstraintFilter) Source(opt opt.Options, store *core.StoreInfo) bool {
+	return !placement.MatchLabelConstraints(store, f.constraints)
+}
+
+// Target filters stores when select them as schedule target.
+func (f labelConstraintFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
+	return !placement.MatchLabelConstraints(store, f.constraints)
+}
+
+// RegionFitter is the interface that can fit a region against placement rules.
+type RegionFitter interface {
+	FitRegion(*core.RegionInfo) *placement.RegionFit
+}
+
+type ruleFitFilter struct {
+	scope   string
+	fitter  RegionFitter
+	region  *core.RegionInfo
+	oldFit  *placement.RegionFit
+	oldPeer uint64
+}
+
+// NewRuleFitFilter creates a filter that ensures after replace a peer with new
+// one, the isolation level will not decrease. Its function is the same as
```

Comment:
```suggestion
// one, the isolation level of target store will not decrease. Its function is the same as
```

Reply:
hmm, it is not the target store's isolation level. The isolation level belongs to a region. It is used to evaluate how a region's peers are isolated.